server/routes/v1/tokenDiscovery.js

```diff
@@ -3,12 +3,23 @@ const rippled = require('../../lib/rippled');
 
 const log = require('../../lib/logger')({ name: 'token discovery' });
 
-const TIME_INTERVAL = 1000 * 60 * 5; // 5 minutes
+// Whether this is running in the prod environment (or in dev/staging)
+// For the purpose of running locally, this equals false if the env var doesn't exist
+// DO NOT SET TO TRUE UNLESS YOU'RE SURE ABOUT BIGQUERY USAGE
+// aka don't let the site run after you're done using it, because it'll cost $$
+const IS_PROD_ENV = process.env.REACT_APP_MAINNET_LINK
+  ? process.env.REACT_APP_MAINNET_LINK.includes('xrpl.org')
+  : false;
+// How long the auto-caching should run in dev and staging environments
+// We want to turn it off after some time so it doesn't run when we don't need it, which costs us
+// money per BigQuery query
+const TIME_TO_TEST = 1000 * 60 * 60 * 1; // 1 hour
+
+const TIME_INTERVAL = 1000 * 60 * 30; // 30 minutes
```

Comment:
```suggestion
const TIME_INTERVAL = 1000 * 60 * 180; // 180 minutes
```

I propose we make this call every 3 hours instead. After the changes pushed on 10/29/21, we've reduced our BQ costs from $2,822 per day to now around $500/day (i.e., by reducing the # of servers each calling BQ every 5 mins). However, this is still too costly in dollar terms, and 180 mins still feels reasonable from a UX perspective. Thoughts?

Reply:
We'll move it down another 6x for 30 minutes (so $80ish/day). Is that still too much in your opinion?